packages/chains-deploy/src/setupNetworks.ts

```diff
@@ -153,9 +160,25 @@ export async function setupNetworksForEnv() {
 
         console.debug(`🛠️  Deploying ${network.name}`);
 
+        //TODO: get mailbox address
+        //TODO: Is the slug name === hyperlane name??? Add custom override for when these don't match
+        const hyperlaneChain = Object.values(hyperlaneMetadata).find((hyperlaneChain) => {
+            const chainId =
+                typeof hyperlaneChain.chainId === "number" ? hyperlaneChain.chainId : parseInt(hyperlaneChain.chainId);
+            return chainId === chain.id;
+        });
+        const hyperlaneChainName = hyperlaneChain?.name;
+
+        const hyperlaneAddresses = hyperlaneChainName
+            ? await getHyperlaneRegistryChainAddresses(hyperlaneChainName)
```

Comment:
Might be more efficient to just get all addresses before loop instead of one await per loop iteration